### ISSUE TITLE
What is the best Procedure to add/Train BackGround Images to a Model that is working very Well

### ISSUE THREAD

**Issue by DawaraGit** [ultralytics/yolov5#12073]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi , I have a Model trainned and working very good with the backgrounds I include in the training.
When I move/install My "Detection System" to a new scenario , some fixed objects are detected and I want to prevent that , retraining my Model to "Teach the Model" about those "fixed/static objects or Shadows" that are sometimes similar or looks like a person .
Said that, I make a test with a training using my old model as source and adding 95% of new BackGrounds pics (1000 jpg with no labels ) and a few new pics with Labels to prevent errors, and when testing , the new model is not working "Better", on the contrary is Detecting Worse or less than my old or Original Model.

What is the right way to do that ??? Starting all over again from scratch and Train my Model adding thos new BackGround images ???

Thanks in Advance !!!!!!!!!!!!!!!!

Dawara

**Comment by glenn-jocher** [comment]
@DawaraGit hi Dawara,

When you want to add new background images to your trained model, the best approach is to fine-tune the model using the new dataset. Instead of training from scratch, you can use your old model as the starting checkpoint and continue training on the new dataset.

In your case, it seems like the new model is performing worse than the original model. This can be due to several reasons such as insufficient training data, imbalance in the dataset, or the importance of the new background images relative to the original images. To improve the performance of your model, you may need to adjust the training process by adding more labeled images that specifically cover the scenarios where fixed objects are being detected.

I would recommend starting with a smaller number of new background images and gradually increasing the number as you see improvements in the model's performance. Additionally, you can experiment with data augmentation techniques, such as adding noise or applying transformations to your new background images, to make the model more robust to variations in the new scenarios.

Remember that training a robust and accurate object detection model is an iterative process, so don't hesitate to experiment with different approaches and fine-tune your model accordingly.

Best regards,
Glenn Jocher

**Comment by DawaraGit** [comment]
@glenn-jocher Thanks so much for your response and your time !!!  I will keep trying  and let you know ... 

Best Regards!!!
DWR

**Comment by glenn-jocher** [comment]
Hi @DawaraGit,

You're welcome! I'm glad to be of help. Definitely, keep experimenting and fine-tuning your model. If you have any further questions or need assistance, feel free to ask. 

Best Regards,
Glenn

**Comment by DawaraGit** [comment]
@glenn-jocher Thanks so much for your responses.

Now I'm having some new scenarios and I would like to share and may be you can give me some light on it 

In April 2022 I run this ....

_!python  train.py --img 448 --batch 32 --epochs 300 --data COCO128MyDataset.yaml  --patience 30 --cache  --weights /yolov5l6.pt_ 

It response with...

Downloading https://ultralytics.com/assets/Arial.ttf to /root/.config/Ultralytics/Arial.ttf...
train: weights=/yolov5l6.pt, cfg=, data=/COCO128MyDataset.yaml, hyp=data/hyps/hyp.scratch-low.yaml, epochs=300, batch_size=32, imgsz=448, rect=False, resume=False, nosave=False, noval=False, noautoanchor=False, evolve=None, bucket=, cache=ram, image_weights=False, device=, multi_scale=False, single_cls=False, optimizer=SGD, sync_bn=False, workers=8, project=runs/train, name=exp, exist_ok=False, quad=False, cos_lr=False, label_smoothing=0.0, patience=30, freeze=[0], save_period=-1, local_rank=-1, entity=None, upload_dataset=False, bbox_interval=-1, artifact_alias=latest
Command 'git fetch && git config --get remote.origin.url' timed out after 5 seconds
YOLOv5 🚀 v6.1-11-g63ddb6f torch 1.10.0+cu111 CUDA:0 (Tesla P100-PCIE-16GB, 16281MiB)

hyperparameters: lr0=0.01, lrf=0.01, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0, copy_paste=0.0
Weights & Biases: run 'pip install wandb' to automatically track and visualize YOLOv5 🚀 runs (RECOMMENDED)
TensorBoard: Start with 'tensorboard --logdir runs/train', view at http://localhost:6006/



Model Summary: 607 layers, 76770172 parameters, 76770172 gradients, 111.5 GFLOPs

remote: Enumerating objects: 6, done.
remote: Counting objects: 100% (4/4), done.
remote: Total 6 (delta 4), reused 4 (delta 4), pack-reused 2
Unpacking objects: 100% (6/6), done.
From https://github.com/ultralytics/yolov5
   d2e698c..b804b36  master     -> origin/master
Transferred 795/795 items from /yolov5l6.pt
Scaled weight_decay = 0.0005
optimizer: SGD with parameter groups 131 weight (no decay), 135 weight, 135 bias
albumentations: version 1.0.3 required by YOLOv5, but version 0.1.12 is currently installed
train: Scanning 'labels' images and labels...10540 found, 1030 missing, 3 empty, 7 corrupt: 100% 11572/11572 [27:55<00:00,  6.91it/s] 


train: Caching images (4.6GB ram): 100% 11565/11565 [03:33<00:00, 54.11it/s]
val: Scanning '/DataSet4Val/labels.cache' images and labels... 1202 found, 1 missing, 0 empty, 0 corrupt: 100% 1203/1203 [00:00<?, ?it/s]
val: Caching images (0.5GB ram): 100% 1203/1203 [00:38<00:00, 31.44it/s] 
Plotting labels to runs/train/expXX/labels.jpg... 

Image sizes 448 train, 448 val
Using 4 dataloader workers
Logging results to runs/train/exp56
Starting training for 300 epochs...

     Epoch   gpu_mem       box       obj       cls    labels  img_size
     0/299     10.3G   0.06297   0.02404  0.004835        61       448: 100% 362/362 [07:50<00:00,  1.30s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 19/19 [00:14<00:00,  1.30it/s]
                 all       1203       2507      0.482      0.592      0.461      0.277

     Epoch   gpu_mem       box       obj       cls    labels  img_size
     1/299     11.5G   0.05435    0.0219   0.00194        79       448: 100% 362/362 [07:42<00:00,  1.28s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 19/19 [00:14<00:00,  1.31it/s]
                 all       1203       2507       0.55      0.553      0.477       0.27

     Epoch   gpu_mem       box       obj       cls    labels  img_size
     2/299     11.5G   0.05596   0.02251  0.002191        50       448: 100% 362/362 [07:38<00:00,  1.27s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 19/19 [00:14<00:00,  1.31it/s]
                 all       1203       2507      0.572      0.489      0.456      0.241

.....

    Epoch   gpu_mem       box       obj       cls    labels  img_size
   100/299     11.7G   0.04132   0.01705 0.0006751        82       448: 100% 362/362 [07:47<00:00,  1.29s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 19/19 [00:15<00:00,  1.19it/s]
                 all       1203       2507      0.505      0.601      0.458      0.277

     Epoch   gpu_mem       box       obj       cls    labels  img_size
   101/299     11.7G   0.04128   0.01711 0.0006663        79       448: 100% 362/362 [07:46<00:00,  1.29s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 19/19 [00:15<00:00,  1.19it/s]
                 all       1203       2507      0.504      0.604      0.458      0.277

     Epoch   gpu_mem       box       obj       cls    labels  img_size
   102/299     11.7G   0.04106   0.01694 0.0006846        92       448: 100% 362/362 [07:46<00:00,  1.29s/it]
               Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 100% 19/19 [00:15<00:00,  1.19it/s]
                 all       1203       2507      0.508      0.598      0.457      0.277
Stopping training early as no improvement observed in last 30 epochs. Best results observed at epoch 72, best model saved as best.pt.
To update EarlyStopping(patience=30) pass a new patience value, i.e. `python train.py --patience 300` or use `--patience 0` to disable EarlyStopping.

41 epochs completed in 5.522 hours.  _*( It was retrainned )_
Optimizer stripped from runs/train/expXX/weights/last.pt, 154.1MB
Results saved to runs/train/expXX

**SO , THATS my BEST Model ... IT Works great !!**

**NEXT CHAPTER !!**
This Week ( September 2023 ) , Im trying to train it again, I mean, from same ckeckpoint model , exactly the same custom dataset with 11572 images , same Validation Dataset ... and with Yolov5 v7 and with YOLOv8 I am getting this:

**Yolov5**
!python train.py --img 448 --batch 32 --epochs 300 --data /COCO128MyDataset.yaml  --patience 30 --cache --weights yolov5l6.pt

Stopping training early as no improvement observed in last 30 epochs. Best results observed at epoch 1, best model saved as best.pt.
To update EarlyStopping(patience=30) pass a new patience value, i.e. `python train.py --patience 300` or use `--patience 0` to disable EarlyStopping.

32 epochs completed in 0.594 hours.
Optimizer stripped from runs/train/exp/weights/last.pt, 154.1MB
Optimizer stripped from runs/train/exp/weights/best.pt, 154.1MB

Validating runs/train/exp/weights/best.pt...
Fusing layers... 
Model summary: 346 layers, 76726332 parameters, 0 gradients, 111.2 GFLOPs
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100%|██████████| 19/19 [00:11<00:00,  1.68it/s]
                   all       1203       2507      0.554      0.562      0.496      0.295
                person       1203       2507      0.554      0.562      0.496      0.295
Results saved to runs/train/exp


**YOLOv8**
!yolo train data=COCO128MyDataset.yaml model=yolov8l.pt epochs=300 lr0=0.01 cache=true imgsz=448
 Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
     51/300      5.77G      1.273     0.9391      1.274         48        448: 100% 723/723 [01:33<00:00,  7.75it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 38/38 [00:05<00:00,  7.06it/s]
                   all       1203       2507      0.508      0.561      0.464       0.29
Stopping training early as no improvement observed in last 50 epochs. Best results observed at epoch 1, best model saved as best.pt.
To update EarlyStopping(patience=50) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping.

51 epochs completed in 1.424 hours.
Optimizer stripped from runs/detect/train4/weights/last.pt, 87.7MB
Optimizer stripped from runs/detect/train4/weights/best.pt, 87.7MB

Validating runs/detect/train4/weights/best.pt...
Ultralytics YOLOv8.0.170 🚀 Python-3.10.12 torch-2.0.1+cu118 CUDA:0 (NVIDIA A100-SXM4-40GB, 40514MiB)
Model summary (fused): 268 layers, 43668288 parameters, 0 gradients
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 38/38 [00:04<00:00,  8.20it/s]
                   all       1203       2507      0.591      0.548      0.523      0.341
                person       1203       2507      0.591      0.548      0.523      0.341
Speed: 0.0ms preprocess, 1.0ms inference, 0.0ms loss, 0.7ms postprocess per image


a new RUN....
!yolo train data=/COCO128MyDataset.yaml model=yolov8l.pt epochs=300  cache=true patience=5

  Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      6/300      11.2G      1.603      1.465      1.573         81        640: 100% 723/723 [01:55<00:00,  6.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 38/38 [00:06<00:00,  5.91it/s]
                   all       1203       2507      0.469      0.461      0.371      0.198
Stopping training early as no improvement observed in last 5 epochs. Best results observed at epoch 1, best model saved as best.pt.
To update EarlyStopping(patience=5) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping.

6 epochs completed in 0.210 hours.
Optimizer stripped from runs/detect/train6/weights/last.pt, 87.7MB
Optimizer stripped from runs/detect/train6/weights/best.pt, 87.7MB

Validating runs/detect/train6/weights/best.pt...
Ultralytics YOLOv8.0.170 🚀 Python-3.10.12 torch-2.0.1+cu118 CUDA:0 (NVIDIA A100-SXM4-40GB, 40514MiB)
Model summary (fused): 268 layers, 43668288 parameters, 0 gradients
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 38/38 [00:05<00:00,  6.67it/s]
                   all       1203       2507      0.544      0.565      0.511      0.318
                person       1203       2507      0.544      0.565      0.511      0.318
Speed: 0.1ms preprocess, 1.7ms inference, 0.0ms loss, 0.7ms postprocess per image


**AND WHEN TESTING** ( always with same Testing images ), the Model Performance is not the same as my BEST model from april 2022.   ( As it has no training It detects worse on my testing images )

I CANT Figure out whats going on .  I Assume the torch ,  CUDA , and also Yolo files  are not the same versions, but Why training is NOT Running anymore ???

As you can see In April 2022 Best results observed at epoch 72 and that was great !

Once Again, just to explain myself :
Same DataSet with 11572 images , same Validation Dataset and same 582 testing images .
I Used the yolov5l6.pt model as checkpoint on april 2022, and now I used the new yolov8l.pt

Whats wrong ??? Or where I must look at ??? 


**My last try : Without checkpoint model with same results .... !!**

!yolo train data=/content/drive/MyDrive/Trainer/COCO128dATAsET4_SIMIL.yaml  epochs=300  cache=true patience=5

 Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
      6/300      3.77G      1.701      1.616      1.534         81        640: 100% 723/723 [01:48<00:00,  6.67it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 38/38 [00:05<00:00,  7.54it/s]
                   all       1203       2507      0.424      0.436      0.325      0.169
Stopping training early as no improvement observed in last 5 epochs. Best results observed at epoch 1, best model saved as best.pt.
To update EarlyStopping(patience=5) pass a new patience value, i.e. `patience=300` or use `patience=0` to disable EarlyStopping.

6 epochs completed in 0.193 hours.
Optimizer stripped from runs/detect/train7/weights/last.pt, 6.5MB
Optimizer stripped from runs/detect/train7/weights/best.pt, 6.5MB

Validating runs/detect/train7/weights/best.pt...
Ultralytics YOLOv8.0.170 🚀 Python-3.10.12 torch-2.0.1+cu118 CUDA:0 (NVIDIA A100-SXM4-40GB, 40514MiB)
Model summary (fused): 168 layers, 3151904 parameters, 0 gradients
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 38/38 [00:04<00:00,  8.52it/s]
                   all       1203       2507      0.499      0.489      0.433      0.248
                person       1203       2507      0.499      0.489      0.433      0.248
Speed: 0.1ms preprocess, 0.4ms inference, 0.0ms loss, 0.6ms postprocess per image


**THANKS in ADVANCE !!!!**

**Comment by DawaraGit** [comment]
@glenn-jocher I hope you can review my last comment in order to help me with some guidance !!!

I will really appreciate that ! Im Stuck for the first time with Yolo !
DR

**Comment by glenn-jocher** [comment]
@DawaraGit Hi DR,

Thank you for sharing the detailed information. It seems like there might be a few factors impacting your training results. The transition from YOLOv5 to YOLOv8 can introduce differences in the model configuration and architecture. As a first step, I recommend double-checking the dataset configuration and model hyperparameters to ensure they are matching the requirements for YOLOv8.

When transitioning between YOLO versions, it's important to confirm that the configurations, such as anchor boxes, input image size, and augmentation parameters, are appropriate for the new model. Additionally, reviewing any changes or updates specific to YOLOv8, including any new hyperparameters or training strategies, could be beneficial.

Another aspect to consider is the impact of the training data and its diversity. It's essential to ensure that the dataset adequately covers the target scenarios, including the new scenarios you mentioned earlier, and that the data is balanced across all classes to avoid biases.

Lastly, evaluating the learning rate schedule, optimizer, and convergence criteria might help identify potential areas for refinement in the training process.

I encourage you to carefully review the training setup and consider these aspects as you continue to navigate the transition to YOLOv8. If you encounter any specific challenges or have further questions, feel free to reach out for more targeted assistance.
